R/cnd-handlers.R

```diff
@@ -1,8 +1,136 @@
-#' Establish handlers on the stack
+#' Try an expression with condition handlers
 #'
 #' @description
 #' `r lifecycle::badge("experimental")`
 #'
+#' `try_catch()` establishes handlers for conditions of a given class
+#' (`"error"` `"warning"`, `"message"`, ...). Handlers are functions
+#' that take a condition object as argument and are called when the
+#' corresponding condition class has been signalled.
+#'
+#' A condition handler can:
+#'
+#' -   *Recover from conditions** with a value. In this case the computation of
+#'     `expr` is aborted and the recovery value is returned from
+#'     `try_catch()`. Error recovery is useful when you don't want
+#'     errors to abruptly interrupt your program but resume at the
+#'     catching site instead.
+#'
+#'     ```
```

Comment:
I tweaked these to not run, so that the code is correctly indented (this must be a roxygen2 bug). If you think the output is really important, you could just manually simulate it.